src/main/java/com/example/onlinebookstore/validation/PasswordValidator.java

```diff
@@ -0,0 +1,15 @@
+package com.example.onlinebookstore.validation;
+
+import com.example.onlinebookstore.dto.user.UserRegistrationRequestDto;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+public class PasswordValidator implements ConstraintValidator<FieldMatch, Object> {
```

Comment:
You can parameterize it with UserRegistrationRequestDto instead of Object to avoid casting.